commit 7990f4d3197c811bee525cc50000e0b1f9e85a4e
Author: bertin <macportsuser@macports.org>
Date:   Sun May 25 22:21:47 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-ignore-empty-env_array.diff

diff --git a/src/port1.0/portutil.tcl b/src/port1.0/portutil.tcl
index bac025bad..b48ac479b 100644
--- a/src/port1.0/portutil.tcl
+++ b/src/port1.0/portutil.tcl
@@ -895,10 +895,14 @@ proc append_to_environment_value {command key args} {
 # debugging purposes.
 proc environment_array_to_string {environment_array} {
     upvar 1 ${environment_array} env_array
-    foreach {key value} [array get env_array] {
-        lappend env_list $key='$value'
+    if {[array size env_array] > 0} {
+        foreach {key value} [array get env_array] {
+            lappend env_list $key='$value'
+        }
+        return "\n[join [lsort $env_list] "\n"]"
+    } else {
+        return ""
     }
-    return "\n[join [lsort $env_list] "\n"]"
 }
 
 ########### Distname utility functions ###########
